src/jose_server.erl

```diff
@@ -718,33 +718,33 @@ rsa_ciphertext(rsa1_5) ->
 	>>;
 rsa_ciphertext(rsa_oaep) ->
 	<<
-		16#B8, 16#F7, 16#0C, 16#A8, 16#F8, 16#30, 16#2A, 16#E9,
-		16#68, 16#8A, 16#DB, 16#3E, 16#5D, 16#AE, 16#84, 16#A7,
-		16#16, 16#FA, 16#9D, 16#E2, 16#FC, 16#81, 16#F7, 16#DF,
-		16#A8, 16#DB, 16#8F, 16#4F, 16#92, 16#A1, 16#51, 16#9E,
-		16#6B, 16#C5, 16#36, 16#CE, 16#93, 16#10, 16#11, 16#D9,
-		16#D5, 16#C2, 16#C9, 16#85, 16#14, 16#EF, 16#D5, 16#C3,
-		16#AC, 16#63, 16#BE, 16#49, 16#FA, 16#02, 16#1A, 16#FC,
-		16#3D, 16#D0, 16#2C, 16#83, 16#C5, 16#76, 16#1D, 16#F5,
-		16#FA, 16#A0, 16#D7, 16#42, 16#ED, 16#3F, 16#A4, 16#12,
-		16#32, 16#14, 16#93, 16#51, 16#79, 16#2E, 16#40, 16#FB,
-		16#14, 16#18, 16#DF, 16#30, 16#62, 16#9F, 16#F3, 16#59,
-		16#5D, 16#83, 16#0F, 16#4A, 16#8F, 16#9B, 16#3F, 16#39
+                16#AF, 16#A4, 16#46, 16#77, 16#B7, 16#39, 16#3A, 16#AB, 
+                16#5A, 16#0C, 16#AB, 16#C2, 16#AD, 16#FA, 16#2C, 16#93, 
+                16#1C, 16#A9, 16#59, 16#46, 16#D3, 16#0A, 16#90, 16#7E, 
+                16#9B, 16#A0, 16#0D, 16#B4, 16#CF, 16#5C, 16#A6, 16#99, 
+                16#C3, 16#71, 16#33, 16#FE, 16#31, 16#12, 16#2C, 16#C2, 
+                16#50, 16#4B, 16#85, 16#D7, 16#45, 16#4E, 16#F2, 16#99, 
+                16#45, 16#95, 16#C8, 16#BA, 16#6E, 16#66, 16#79, 16#31, 
+                16#8E, 16#B7, 16#53, 16#06, 16#20, 16#AC, 16#EB, 16#B6, 
+                16#A6, 16#79, 16#9F, 16#F7, 16#5E, 16#89, 16#E1, 16#8A, 
+                16#81, 16#60, 16#A7, 16#79, 16#94, 16#54, 16#EF, 16#E7, 
+                16#8B, 16#33, 16#7D, 16#1F, 16#E5, 16#AD, 16#1A, 16#66, 
+                16#22, 16#90, 16#2A, 16#49, 16#1C, 16#A3, 16#E6, 16#62
 	>>;
 rsa_ciphertext(rsa_oaep_256) ->
 	<<
-		16#09, 16#24, 16#EA, 16#EB, 16#D4, 16#EF, 16#00, 16#BE,
-		16#8E, 16#02, 16#BE, 16#25, 16#24, 16#24, 16#18, 16#81,
-		16#8D, 16#7A, 16#A2, 16#EB, 16#F1, 16#BE, 16#5C, 16#DC,
-		16#D0, 16#71, 16#43, 16#09, 16#53, 16#12, 16#44, 16#AD,
-		16#8A, 16#CD, 16#F8, 16#45, 16#7F, 16#1F, 16#30, 16#B6,
-		16#54, 16#8E, 16#AB, 16#D2, 16#10, 16#14, 16#BC, 16#CE,
-		16#7A, 16#99, 16#DC, 16#A6, 16#8D, 16#16, 16#5A, 16#A0,
-		16#50, 16#3A, 16#93, 16#0E, 16#53, 16#4A, 16#B5, 16#6B,
-		16#51, 16#E8, 16#43, 16#8F, 16#BD, 16#2D, 16#E0, 16#63,
-		16#36, 16#24, 16#5B, 16#8D, 16#DD, 16#98, 16#AC, 16#37,
-		16#7C, 16#16, 16#DB, 16#03, 16#C8, 16#BD, 16#22, 16#D2,
-		16#15, 16#98, 16#91, 16#B7, 16#3C, 16#01, 16#CF, 16#0E
+                16#6F, 16#97, 16#8F, 16#91, 16#73, 16#78, 16#5E, 16#53, 
```

Comment:
Demonstration that the ciphertext change here is actually encoded using PKCS#1 Version 1.5 padding:

```elixir
private_key = {:RSAPrivateKey, :"two-prime", 1163626716254914564980136459395684807695125169328499377661515176133500664726004569406515061633752264725167480979816271393911346667208692357841077684025487798285456581385497985022555400834820919818460489376858005605167466539406190291, 65537, 837924159180587806355304330809474848246884082520101181469702392652521428664942454552415662964152625679766070004432092615777291304448520734093291934648299519439752355437698350307876611105690193137070602690197584267484734283766402649, 38384139366830395481042688957876838317068837995135971911011570081580402662762585413480764252529057808049699572389029, 30315300419643147991826027205656337561558478820966564595877651670534732295504135676718770733184467217297653615533079, 8179698344403211976261382028254389488932715739812151665611602419386787670905629917217333011136012044298977741245177, 22006659100424535235227783455286330767217672916176881923934255126503957901011783493597441409146757219020337683430523, 31177545157306276284676329900115896567585136566839161660319287166242766583148091022419866881250561784948462027640129, :asn1_NOVALUE}
public_key = {:RSAPublicKey, 1163626716254914564980136459395684807695125169328499377661515176133500664726004569406515061633752264725167480979816271393911346667208692357841077684025487798285456581385497985022555400834820919818460489376858005605167466539406190291, 65537}
ciphertext = <<111,151,143,145,115,120,94,83,5,152,214,198,17,56,93,198,155,198,146,222,178,249,95,203,35,41,147,27,109,126,56,67,91,127,82,175,24,226,247,102,224,156,95,64,116,93,187,91,156,221,152,146,85,203,209,28,224,161,31,31,252,110,61,204,89,20,59,193,29,119,7,26,88,54,188,254,98,131,185,67,113,172,222,106,198,128,62,227,0,117,181,195,73,252,211,15>>

# PKCS#1 Version 2.2 OAEP-SHA-256 decrypt (this should work)
:error = :jose_jwa_pkcs1.decrypt_private(ciphertext, private_key, [{:rsa_padding, :rsa_pkcs1_oaep_padding}, {:rsa_oaep_md, :sha256}])
:error = :public_key.decrypt_private(ciphertext, private_key, [{:rsa_padding, :rsa_pkcs1_oaep_padding}, {:rsa_oaep_md, :sha256}])

# PKCS#1 Version 1.5 decrypt (this should fail)
"ciphertext" = :jose_jwa_pkcs1.decrypt_private(ciphertext, private_key, [{:rsa_padding, :rsa_pkcs1_padding}])
"ciphertext" = :public_key.decrypt_private(ciphertext, private_key, [{:rsa_padding, :rsa_pkcs1_padding}])
```